src/common/registry/fetch/utils/batches.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { makeBatches } from './batches';
+
+describe('makeBatches util', () => {
+  test('should create batches with correct offset and batchSize', () => {
```

Comment:
we could name this case as 'total Amount Not Fully Divisible by Batch Size'

Reply:
and add normal case for fully divisible

Reply:
Added another test for that